cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowNodeState.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import javax.annotation.Nullable;
+
+/**
+ * Class to represent the state of the node in the {@link Workflow}.
+ */
+public final class WorkflowNodeState {
+
+  /**
+   * Status of the node running inside the Workflow.
+   */
+  public enum NodeStatus {
+    RUNNING,
+    COMPLETED,
+    FAILED,
+    KILLED
+  }
+
+  private final String nodeId;
+  private final NodeStatus nodeStatus;
+  private final String runId;
+  private final String failureCause;
+
+  /**
+   * Create a new instance.
+   * @param nodeId id of the node inside the Workflow
+   * @param nodeStatus status of the node
+   * @param runId run id assigned to the node, null if current node represents custom action or predicate
+   * @param failureCause cause of failure, null if execution of the node succeeded
```

Comment:
So what does it carry? Is it just the error message? Why can't we provide the actual cause to the user since this object is part of the programatic API that the user can use to retrieve node states through the WorkflowContext, right? 

I understand that currently we use the same class to serialize the information and store it in the MDS. However, it may not be the right thing to do. Also, there is a way to serialize and deserialize a Throwable (for information purpose, not really reconstructing the real stacktrace, since it's not possible). See `LogThrowable` and `LogThrowableCodec` in twill.